controller/controller.go

```diff
@@ -390,14 +390,18 @@ func ScaleUp(
 		"scaleUpLimit": scaleUpLimit,
 	})
 
-	configYAML, _ := models.NewConfigYAML(scheduler.YAML)
+	configYAML, err := models.NewConfigYAML(scheduler.YAML)
+	if err != nil {
+		return fmt.Errorf("failed to load scheduler configuration: %w", err)
+	}
 
-	existPendingPods, err := pendingOrFailedPods(config, redisClient, scheduler.Name, mr)
+	hasNotReadyPods, err := checkNotReadyPods(config, redisClient, scheduler.Name, configYAML.PreventRoomsCreationWithError, mr)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to list pending or failed pods: %w", err)
 	}
-	if existPendingPods {
-		return errors.New("there are pending or failed pods")
+
+	if hasNotReadyPods {
+		return errors.New("scheduler has pods not ready (pending or with error)")
```

Comment:
minor to keep the same signature of the function:
```suggestion
		return errors.New("scheduler has not ready pods (pending or with error)")
```